Tests/test_energy.gd

```diff
@@ -0,0 +1,60 @@
+extends TestBase
+
+var card: CardBase
+
+# @overide
+func before_each():
+	super()
+	_player_energy_component.ignore_cost = false
+	_player_energy_component.energy = 0
+	_player_energy_component.energy_generation = 3
+	_player_energy_component.Max_energy = 4
+
+func test_add_energy():
+	card = load("res://Cards/Resource/Card_Energy.tres") # adds 70 energy
+
+	card.on_card_play(_player, [_player])
+	assert_eq(_player_energy_component.energy, 70)
+
+
+func test_using_energy():
+	_player_energy_component.add_energy(10)
+	card = load("res://Cards/Resource/Card_Damage.tres") # costs 7 by default
+	card.energy.energy_cost = 7
+
+	card.on_card_play(PlayerManager.player, [_enemy])
+
+	assert_eq(_player_energy_component.energy, 3)
+
+func test_no_energy():
```

Comment:
`test_not_enough_energy` would probably be a lot better
missing void return type for all test functions btw